src/views/Timer.vue

```diff
@@ -2,24 +2,24 @@
   <div class="timers">
     <b-container fluid class="grid">
       <b-row ref="timerComponentGrid" v-packery="{ itemSelector: '.packery-item', percentPosition: true }">
+        <timer v-if="componentState.earth.display" :time="worldstate.earthCycle" location="Earth" />
```

Comment:
The order is roughly as follows:

1. Time components. They always have the same size and are displayed as a single line in the compact theme.
2. The rest of compact components (reset and construction). Darvo component can probably be put here too (it can be transformed into a single line).
3. News. This is probably the most important part nobody wants to miss.
4. Most components in whatever order.
5. Bounties. Big blocks with low information-per-space ratio.
6. Huge price lists (sales, Baro). If put in any other place, they will push all the important information out of the screen.

While the theme doesn't rely on this exact order, it provides several benefits.

1. Simple components are always in the same spot. This is especially important for timers as it's often valuable to check how much time is left for Eidolon fight without wasting precious seconds searching where the block teleported this time.
2. News are always in the same spot too, and at the very top.
3. Huge blocks with the least important information are at the bottom and don't push the more important blocks out of the screen.

All of this applies to the rest of the themes. They get statically positioned timers and "sorting by information density". The only benefit specific to the compact theme is probably all "blockless" components being at the top and not being mixed with the rest, to make the page more aesthetically pleasing.

Reply:
Adding new blocks won't wreck my theme. At worst, if someone adds a block before timers, some of the benefits listed above will be lost, but the theme will function just fine. I'd love to make a CSS-only solution, but the masonry layout relies on JS and doesn't respect the `order` CSS property, unfortunately.

Anyway, is there any reason to not follow the order I listed above, now or in the future? Especially considering it provides benefits for all themes, not just compact.

Reply:
@EricSihaoLin Ideally, I'd prefer the compact mode to be a separate checkbox in the theme settings so it can be applied to any of the themes. However, currently there's essentially only one theme ("Night") and its slight variation ("Day"), so I don't feel it's worth the complications at this point.

When themes are fully implemented, I'll look into changing the "Night Compact" theme into something like "Compact Mode" for any theme, if that's something others are interested in, but I'll probably need some help with the UI part.

"Compact Mode" for every block individually (IIRC, that's what was suggested in Discord) is an option too, but a full compact theme/mode would still be needed as it applies changes outside block contents and numerous toggle buttons use valuable space too.

Reply:
themes are fully implemented, just no one cares about the other 2, and day has no specific colors set >.<